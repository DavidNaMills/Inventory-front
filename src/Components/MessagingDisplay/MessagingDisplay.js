import React from 'react';
import classes from './messagingDisplay.module.scss';
import { useSelector } from 'react-redux';
import ErrorIcon from '@material-ui/icons/Error';
import CheckIcon from '@material-ui/icons/Check';


const MessagingDisplay = () => {
    const { display } = useSelector(state => state);

    return (
        display &&
        <div className={display.messageType === 'errType' ? classes.messagingDisplay__errStyle : classes.messagingDisplay__succStyle}>
            {display.messageType === 'errType' ? <ErrorIcon/> : <CheckIcon/>} <p className={classes.messagingDisplay_p}>{display.message}</p>
        </div>
    )
}

export default MessagingDisplay;