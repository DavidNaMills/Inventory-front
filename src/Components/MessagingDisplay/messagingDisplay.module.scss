@import '../../styles/variables/colors.module.scss';
@import '../../styles/mixins/messageDisplayMixin.module.scss';

.messagingDisplay{
    &__errStyle {
        @include messageDisplay($message-error-back, $light-text);
    }
    &__succStyle{
        @include messageDisplay($submit-shadow, $light-text);        
    }
    
    &_p{
        font-size: 22px;
        padding-left: 10px;
    }
}