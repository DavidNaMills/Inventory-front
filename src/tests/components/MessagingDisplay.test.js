import React from 'react';
import '../testHelpers/context';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

jest.mock('react-redux');
import { useSelector } from 'react-redux';

import MessagingDisplay from '../../Components/MessagingDisplay/MessagingDisplay';

import ErrorIcon from '@material-ui/icons/Error';
import CheckIcon from '@material-ui/icons/Check';


describe('MessagingDisplay', ()=>{
    describe('basic creation tests', ()=>{
        beforeEach(()=>{
            useSelector.mockImplementation(()=>({
                display: {
                    messageType: 'errType',
                    message: 'hello world'
                }
            }));
        });

        it('matches snapshot', ()=>{
            const comp = renderer.create(<MessagingDisplay />)
            const app = comp.toJSON();
            expect(app).toMatchSnapshot();
        });
        
        it('exists', ()=>{
            const wrapper = shallow(<MessagingDisplay />);
            expect(wrapper.exists).toBeTruthy();
        });
    });
        
    describe('error type messages', ()=>{
        let wrapper;
        let mockMsg = 'horribly wrong';

        beforeEach(()=>{
            useSelector.mockImplementation(()=>({
                display: {
                    messageType: 'errType',
                    message: mockMsg 
                }
            }));
            wrapper = mount(<MessagingDisplay />);
        });

        it('renders a <div/> with className = "messagingDisplay__errStyle"', ()=>{
            const div = wrapper.find('div');
            expect(div.length).toBe(1);
            expect(div.props()).toHaveProperty('className', 'messagingDisplay__errStyle');
        });
        
        it('renders a <p> element with the message. element has className: "messagingDisplay_p" ', ()=>{
            const p = wrapper.find('p');
            expect(p.length).toBe(1);
            expect(p.props()).toHaveProperty('className', 'messagingDisplay_p');
            expect(p.text()).toEqual(mockMsg);
        });
        
        it('renders a <ErrorIcon/> icon component ', ()=>{
            expect(wrapper.find(ErrorIcon).length).toBe(1);
        });
    });
    
    describe('success type messages', ()=>{
        let wrapper;
        let mockMsg = 'looking good';

        beforeEach(()=>{
            useSelector.mockImplementation(()=>({
                display: {
                    messageType: 'succType',
                    message: mockMsg 
                }
            }));
            wrapper = mount(<MessagingDisplay />);
        });

        it('renders a <div/> with className = "messagingDisplay__succStyle"', ()=>{
            const div = wrapper.find('div');
            expect(div.length).toBe(1);
            expect(div.props()).toHaveProperty('className', 'messagingDisplay__succStyle');
        });
        
        it('renders a <p> element with the message. element has className: "messagingDisplay_p" ', ()=>{
            const p = wrapper.find('p');
            expect(p.length).toBe(1);
            expect(p.props()).toHaveProperty('className', 'messagingDisplay_p');
            expect(p.text()).toEqual(mockMsg);
        });
        
        it('renders a <CheckIcon/> icon component ', ()=>{
            expect(wrapper.find(CheckIcon).length).toBe(1);
        });
    });

    it('doesnt render anything if no message within the display state', ()=>{
        useSelector.mockImplementation(()=>({
            display: null
        }));
        const wrapper = mount(<MessagingDisplay />);
        expect(wrapper.exist).toBeFalsy();
    });
});