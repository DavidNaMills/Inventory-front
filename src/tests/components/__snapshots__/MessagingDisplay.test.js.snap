// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagingDisplay basic creation tests matches snapshot 1`] = `
<div
  className="messagingDisplay__errStyle"
>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
    />
  </svg>
   
  <p
    className="messagingDisplay_p"
  >
    hello world
  </p>
</div>
`;
